src/commit_log.rs

```diff
@@ -119,8 +122,76 @@ pub trait CommitLog {
     fn scoped_subscribe<'a>(
         &'a self,
         consumer_group_name: &str,
-        offsets: Option<Vec<ConsumerOffset>>,
+        offsets: Vec<ConsumerOffset>,
         subscriptions: Vec<Subscription>,
         idle_timeout: Option<Duration>,
     ) -> Pin<Box<dyn Stream<Item = ConsumerRecord> + Send + 'a>>;
 }
+
+fn nullable_vec<'de, D, T>(d: D) -> Result<Vec<T>, D::Error>
+where
+    D: Deserializer<'de>,
+    T: Deserialize<'de>,
+{
+    Deserialize::deserialize(d).map(|x: Option<_>| x.unwrap_or_default())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_nullable_vec_handles_null() {
+        let json = r#"
```

Comment:
Not sure if I already commented on this (somewhere else), but you could use the `json!` macro from serde_json to create JSON values instead of raw strings. The benefit is that the compiler validates that the JSON value is correct.

Reply:
Yeah, I will move towards that. Thanks.